store/sqlstore/post_store.go

```diff
@@ -2252,7 +2252,7 @@ func (s *SqlPostStore) cleanupThreads(postId, rootId string, permanent bool) err
 		return nil
 	}
 	if rootId != "" {
-		thread, err := s.Thread().Get(rootId)
+		thread, err := s.Thread().Get(WithMaster(context.Background()), rootId)
 		if err != nil {
 			var nfErr *store.ErrNotFound
 			if !errors.As(err, &nfErr) {
```

Comment:
GH won't let me comment on the lines below but this is about the Get and Update queries combined. It seems like we are fetching the thread object by its ID, setting replyCount to `-1` and then writing it back to the DB again. Any reason we cannot do a simple `UPDATE threads set replyCount = -1 where id = <>`?

Reply:
I believe we need to decrement the count by 1 there. Either way you are correct, it can probably be done with a single query:

```sql
UPDATE Threads 
SET ReplyCount = ReplyCount - 1
WHERE PostId = ?
AND ReplyCount > 0
```

Reply:
lol yeah .. that would have been a funny bug

Reply:
Created [MM-36366](https://mattermost.atlassian.net/browse/MM-36366)